PocketKit/Sources/PocketKit/MyList/SearchItemsList/SearchViewModel.swift

```diff
@@ -64,14 +69,26 @@ class SearchViewModel: ObservableObject {
         let term = searchTerm.trimmingCharacters(in: .whitespaces).lowercased()
         let shouldShowUpsell = !isPremium && selectedScope == .all
         guard !shouldShowUpsell else { return }
-
-        emptyState = searchResultState()
+        submitSearch(with: term)
         showRecentSearches = false
 
         guard isPremium, !term.isEmpty else { return }
         recentSearches = updateRecentSearches(with: term)
     }
 
+    func clear() {
+        self.searchResults = []
+    }
+
+    private func submitSearch(with term: String) {
+        clear()
+        source.search(for: term, scope: selectedScope)
+        source.events.receive(on: DispatchQueue.main).sink { [weak self] event in
+            guard case let .searchListFetched(items) = event, let self else { return }
+            self.searchResults = items.compactMap { SearchItem(item: $0) }
+        }.store(in: &subscriptions)
```

Comment:
Maybe I'm overlooking something - is this subscription ever getting cancelled? E.g when the search completed, else we have a lingering cancellable every time a search is performed.

Reply:
Nope I overlooked this! You're correct, I'll clear out the subscription in `clear()`